frame/assets/src/lib.rs

```diff
@@ -1040,18 +1041,21 @@ pub mod pallet {
 			Asset::<T, I>::try_mutate(id, |maybe_details| {
 				let details = maybe_details.as_mut().ok_or(Error::<T, I>::Unknown)?;
 				ensure!(details.status == AssetStatus::Live, Error::<T, I>::LiveAsset);
-				ensure!(origin == details.owner, Error::<T, I>::NoPermission);
-				if details.owner == owner {
+				ensure!(Some(origin) == details.owner.clone(), Error::<T, I>::NoPermission);
+
+				if details.owner == Some(owner.clone()) {
 					return Ok(())
 				}
 
 				let metadata_deposit = Metadata::<T, I>::get(id).deposit;
 				let deposit = details.deposit + metadata_deposit;
 
 				// Move the deposit to the new owner.
-				T::Currency::repatriate_reserved(&details.owner, &owner, deposit, Reserved)?;
+				if let Some(details_owner) = &details.owner {
+					T::Currency::repatriate_reserved(details_owner, &owner, deposit, Reserved)?;
+				}
```

Comment:
I think we should _try_ to `unreserve` the amount from the current owner, and then `reserve` it from the new owner. One, this is causing a balance transfer. Two, the deposit costs may have changed since the asset was created. If they have gone up, and the current owner doesn't have enough reserved, it will fail (and the owner will be stuck unless they reserve more via some other operation). Since we've calculated the going rate for deposits here, the new owner should have to reserve the updated amount.